### PR TITLE
Support altering column comments in Hive Glue catalogs

### PR DESCRIPTION
**Pull request by alexjo2144**
## Description

Support altering column comments in Hive Glue catalogs.

## Additional context and related issues

Extracted from: https://github.com/trinodb/trino/pull/19073

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
# Hive
* Support altering column comments in Hive Glue catalogs.
```

### CODE REVIEW

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java [hunk 1]
```diff
@@ -727,7 +727,26 @@ public void setTableOwner(String databaseName, String tableName, HivePrincipal p
...
+                .build();
+        replaceTable(databaseName, tableName, newTable, null);
+    }
+
+    private static List<Column> updateColumnComment(List<Column> columns, String columnName, Optional<String> comment)
+    {
+        return columns.stream()
+                .map(column -> {
+                    if (column.getName().equals(columnName)) {
+                        return new Column(column.getName(), column.getType(), comment, column.getProperties());
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java:745]: - should the column name matching be case-insensitive?
- `updateColumnComment` should fail when no column matched, and when 2+ matched
   - and we should fail when data and partitioning column both found

**Review by findepi — COMMENTED** [review]
Should this come with any test changes?

**Comment by findepi** [comment]
/test-with-secrets sha=17cb3d1477a8d7891b5916476715a0bbced985bf

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java (hunk 1 above)
Comment (findinpath) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java:745]: `TestHiveGlueMetastore` could potentially host the test

Comment (alexjo2144) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java:745]: > should the column name matching be case-insensitive?

To be extra safe we could but they don't need to be. Glue lowercases everything.

> updateColumnComment should fail when no column matched, and when 2+ matched
> and we should fail when data and partitioning column both found

Can do

Comment (findepi) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java:745]: > To be extra safe we could but they don't need to be. Glue lowercases everything.
> 
> >

right. let's mention this wisdom in a code comment

**Comment by alexjo2144** [comment]
Test and sanity checks added, PTAL

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java [hunk 2]
```diff
@@ -727,7 +728,56 @@ public void setTableOwner(String databaseName, String tableName, HivePrincipal p
...
+        replaceTable(databaseName, tableName, updatedTable, null);
+    }
+
+    private static Optional<Integer> indexOfColumnWithName(List<Column> columns, String columnName)
+    {
+        Optional<Integer> index = Optional.empty();
+        for (int i = 0; i < columns.size(); i++) {
+            // Glue columns are always lowercase
+            if (columns.get(i).getName().equals(columnName)) {
+                index.ifPresent(ignored -> {
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java:759]: ordinary `if (index.isPresent())` would free you from having to name a variable you ignore

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
/test-with-secrets sha=f16036dd4d290ef621025a6b4e0eeb4e531ac620

**Pull request merged**